### PR TITLE
chore: convert babel-preset-gatsby/index and test to ts

### PR DESCRIPTION
**Pull request by lwkchan**
## Description

Converts `babel-preset-gatsby/index.js` and test to TypeScript

### Documentation

n/a

## Related Issues

https://github.com/gatsbyjs/gatsby/issues/21995

### CODE REVIEW

**Comment by blainekasten** [comment]
Thanks for this! Looks like we have some lint errors to fix up to get the CI checks to turn green!

**Comment by lwkchan** [comment]
Thanks @blainekasten - got CI to work now

**Comment by ascorbic** [comment]
Hi @lwkchan 
Unfortunately some conflicts have crept in since you opened this. Is there any chance you could resolve them so we can get this merged?

Thanks

File: packages/babel-preset-gatsby/src/index.ts [hunk 1]
```diff
@@ -29,7 +29,12 @@ export function loadCachedConfig() {
   return pluginBabelConfig
 }
 
-export default function preset(_, options = {}) {
+interface IOptions {
+  targets?: Browserslist | { node: `current` } | string
+  stage?: string
```
Comment (ascorbic) [inline packages/babel-preset-gatsby/src/index.ts:34]: It's failing typechecks because this also needs `reactRuntime?: "classic" | "automatic"`.

**Comment by j0sh77** [comment]
Hey! Thanks so much for opening this pull request!

Sadly this PR got stale and didn't have any activity for some time. We're trying to do better with PR reviews! To get a better overview of all actionable PRs we're going through all open PRs and triage them. Since we won't be able to do everything and adding new features _always_ means added maintenance burden, we have to be more picky about what's beneficial for the average user and the project itself longterm.

We think this is a great PR and would love to see it land in Gatsby. We're closing this PR _for now_ but if you're able to rebase onto the latest `master` branch and try it out with the latest `next` versions, we'd be happy to review the PR when its re-opened.

We absolutely want to have you as a contributor and are sorry for any inconveniences we caused with replying too late to this PR.

Thanks for submitting to Gatsby! :purple_heart:

**Pull request closed without merging**